src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1793,6 +1793,36 @@ inline constexpr auto mutate_apply(
                                       std::forward<Args>(args)...);
 }
 
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Apply the function `F` mutating its `return_tags` and taking as
+ * additional arguments its `argument_tags` and `args`.
+ *
+ * \details
+ * `F` must have `tmpl::list` type aliases `return_tags` and `argument_tags`, as
+ * well as a static `apply` function. The `apply` function must take the types
+ * of the `return_tags` as `gsl::not_null` pointers, followed by the types of
+ * the `argument_tags` as constant references. It can also take the `Args` as
+ * additional arguments.
+ *
+ * \example
+ * \snippet Test_DataBox.cpp mutate_apply_struct_definition_example
+ * This is how to use `mutate_apply` with the above class:
+ * \snippet Test_DataBox.cpp mutate_apply_struct_example_with_tags_lists
+ *
+ * \see box
+ * \tparam F The function to apply
+ * \param box The DataBox out of which to retrieve the Tags to pass to `F`
+ * \param args The arguments to pass to the function that are not in the
+ * DataBox, `box`
+ */
+template <typename F, typename BoxTags, typename... Args>
+inline constexpr auto mutate_apply(const gsl::not_null<DataBox<BoxTags>*> box,
+                                   Args&&... args) {
```

Comment:
`noexcept` (You don't have to do the computed noexcept done in some other places in this file.)